content/projects/gsoc/2020/project-ideas/jenkins-rest-plugin.adoc

```diff
@@ -4,15 +4,65 @@ title: "Jenkins REST Plugin"
+
+In the following, we show how the finished plugin would look like from the user point of view in a Jenkins Pipeline DSL program. This is not a specification, it is only an example. The student is expected to study the the link:https://github.com/jenkinsci/docker-workflow-plugin[jenkins-rest] library, the Jenkins Plugin tutorials and the Scripted Pipeline syntax, and propose a proper Jenkins Pipeline DSL syntax for this project.
+
+=== Quick start
+There are many technologies to use together to form this plugin. The student who wishes to get started will need to:
+
+* Try out Jenkins REST APIs
+* Try writing a small client using one of REST API Client libraries
+* study plugin tutorials on how to write a Pipeline Step plugin
+  - Tutorials listed on the link:https://github.com/jenkinsci/docker-workflow-plugin[student information page]
+  - link:https://github.com/jenkinsci/docker-workflow-plugin[Writing Pipeline compatible plugins]
+  - link:https://github.com/jenkinsci/workflow-step-api-plugin/blob/master/README.md[Writing Pipeline steps]
+  - link:https://github.com/jenkinsci/workflow-step-api-plugin/blob/master/README.md[Updating plugin for Pipeline]
+  - looking at existing pipeline compatible plugins will be very useful. Example:
+      ** link:https://github.com/jenkinsci/workflow-step-api-plugin/blob/master/README.md[External Workspace Manager] (look at the link:https://github.com/jenkinsci/workflow-step-api-plugin/blob/master/README.md[steps folder], the steps themselves, and their execution classes)
+* study the link:https://github.com/jenkinsci/workflow-step-api-plugin/blob/master/README.md[jenkins-rest] library, try the examples in the link:https://github.com/jenkinsci/workflow-step-api-plugin/blob/master/README.md[wiki]
+* create a basic custom pipeline compatible plugin and load it in Jenkins (see the plugin tutorials)
+
+=== Links
+* The two java rest api libraries:
+  - https://github.com/jenkinsci/java-client-api
+  - https://github.com/cdancy/jenkins-rest
+* Mailing list discussion: https://groups.google.com/forum/#!topic/jenkinsci-dev/x-EbjnWcFqs
+* Discussion on REST: https://groups.google.com/forum/#!topic/jenkinsci-dev/mYeM5qA6tGM
+
+=== Skills to study/improve
+* Java
+* REST API
+* Jenkins Pipeline
+
```

Comment:
Wonder if these line feeds are causing problems. Can you only have 1 line feed

Reply:
Yes, I have updated the file with all skills mentioned in one line.